model/src/main/scala/lucuma/schemas/enums/ProposalAttachmentType.scala

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.schemas.enums
+
+import cats.Eq
+import cats.derived.*
+import io.circe.*
+import io.circe.generic.semiauto
+import lucuma.core.util.Display
+import lucuma.core.util.Enumerated
+import monocle.Focus
+import monocle.Lens
+
+case class ProposalAttachmentType(
+  tag:       String,
+  shortName: String,
+  longName:  String
+) derives Eq:
+  val accept: String = ".pdf" // only PDF files are currently allowed
+
+object ProposalAttachmentType:
+  given Display[ProposalAttachmentType] = Display.by(_.shortName, _.longName)
+
+  val tag: Lens[ProposalAttachmentType, String]       = Focus[ProposalAttachmentType](_.tag)
+  val shortName: Lens[ProposalAttachmentType, String] = Focus[ProposalAttachmentType](_.shortName)
+  val longName: Lens[ProposalAttachmentType, String]  = Focus[ProposalAttachmentType](_.longName)
+
+  val values: List[ProposalAttachmentType] =
+    // This is a meta decoder, not a decoder for enum instances (which comes from the `Enumerated` instance)
+    given Decoder[ProposalAttachmentType] = semiauto.deriveDecoder
+
+    DynamicEnums.parsedEnums
```

Comment:
I wonder if we can abstract away the common logic of all dynamic enums.

Reply:
We might be able to. 🤔 I'll probably have to add `Partner` at some point soon. I'll look at it then.